docs/examples/rag_weaviate.ipynb

```diff
@@ -0,0 +1,1092 @@
+        "# Delete the collection if it already exists\n",
+        "if (client.collections.exists(collection_name)):\n",
+        "    client.collections.delete(collection_name)\n",
+        "\n",
+        "# Create the collection\n",
+        "collection = client.collections.create(\n",
+        "    name=collection_name,\n",
+        "    vectorizer_config=wc.Configure.Vectorizer.text2vec_openai(\n",
+        "        model=\"text-embedding-3-large\",                           # Specify your embedding model here\n",
+        "    ),\n",
+        "\n",
+        "    # Enable generative model from Cohere\n",
+        "    generative_config=wc.Configure.Generative.openai(\n",
+        "    model=\"gpt-4o\"                                                # Specify your generative model for RAG here\n",
+        "    ),\n",
+        "\n",
+        "    # Define properties of metadata\n",
+        "    properties=[\n",
+        "        wc.Property(\n",
+        "            name=\"text\",\n",
+        "            data_type=wc.DataType.TEXT\n",
+        "        ),\n",
+        "        wc.Property(\n",
+        "            name=\"title\",\n",
+        "            data_type=wc.DataType.TEXT,\n",
+        "            skip_vectorization=True\n",
+        "        ),\n",
+        "    ]\n",
+        ")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "RgMcZDB9Dzfs"
+      },
+      "source": [
+        "### Wrangle data into an acceptable format for Weaviate\n",
+        "\n",
+        "Transform our data from lists to a list of dictionaries for insertion into our Weaviate collection."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 10,
+      "metadata": {
+        "id": "kttDgwZEsIJQ"
+      },
+      "outputs": [],
+      "source": [
+        "# Initialize the data object\n",
+        "data = []\n",
+        "\n",
+        "# Create a dictionary for each row by iterating through the corresponding lists\n",
+        "for text, title in zip(texts, titles):\n",
+        "    data_point = {\n",
+        "        \"text\": text,\n",
+        "        \"title\": title,\n",
+        "    }\n",
+        "    data.append(data_point)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "-4amqRaoD5g0"
+      },
+      "source": [
+        "### Insert data into Weaviate and generate embeddings\n",
+        "\n",
+        "Embeddings will be generated upon insertion to our Weaviate collection."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "g8VCYnhbaxcz",
+        "outputId": "cc900e56-9fb6-4d4e-ab18-ebd12b1f4201"
+      },
+      "outputs": [],
+      "source": [
+        "# Insert text chunks and metadata into vector DB collection\n",
+        "response = collection.data.insert_many(\n",
```

Comment:
This raises the an error [1] for me (I thought it could be because I am not using a paid OpenAI account, but that is not indicated anywhere in the stacktrace).

Does this error ring any bell? (Perhaps some timeout requiring a bump?)
BTW: how complicated would it be to switch to a setup (embeddings + LLM) that does not require a subscription, e.g. Hugging Face? 🤔 

[1]:
```
WeaviateBatchError: Query call with protocol GRPC batch failed with message <AioRpcError of RPC that terminated with:
	status = StatusCode.DEADLINE_EXCEEDED
	details = "Deadline Exceeded"
	debug_error_string = "UNKNOWN:Error received from peer  {grpc_message:"Deadline Exceeded", grpc_status:4, created_time:"2024-12-10T21:29:45.487853+01:00"}"

```

Reply:
@vagenas Are you using a valid OpenAI API key?

Regarding Hugging Face, Weaviate currently only has functionality to use Hugging Face embedding models but not generative models. The only open source generative module that Weaviate currently supports is Ollama.